### DISCUSSION TITLE
Compile LaTeX on file save

### DISCUSSION THREAD

**Discussion by giacomogallina** [helix-editor/helix#5002] [Q&A] [answered]
I'm sorry if this is a duplicate, but I wasn't able to quickly find a way to compile compile a .tex file to .pdf automatically when saving. More generally I would like to tell Helix to execute a specific command on file save, based on the file language. Is it already possible?

**Comment by the-mikedavis** [comment, marked as answer]
> execute a specific command on file save, based on the file language

This is not currently possible.

From what I understand, the latex language server TexLab can compile and generate PDFs on save though. The documentation link [here](https://github.com/latex-lsp/texlab#requirements) looks broken at the moment but it should be possible for the language server to do this as long as you have a TeX distribution installed

**Reply by cgahr** [reply]
Is this planed as a feature?

**Comment by giacomogallina** [comment]
Thanks! I realized that I had this already configured and somehow I had forgotten about it, sorry to have bothered you!
Anyway, for anyone else who might need this, this `languages.toml` seems to work:
```
[[language]]
name = "latex"
scope = "source.tex"
injection-regex = "tex"
file-types = ["tex"]
roots = []
comment-token = "%"
language-server = { command = "texlab" }
config = { texlab = { build = { onSave = true } } }
indent = { tab-width = 4, unit = "\t" }
```

**Comment by peredwardsson** [comment]
There are outside tools to use for this type of task as well if you prefer, for instance [watchexec](https://github.com/watchexec/watchexec).

**Reply by yjhn** [reply]
[`latexmk`](https://personal.psu.edu/~jcc8/software/latexmk/) with the `-pvc` option can do this too (see [manual](https://manned.org/latexmk.1) for details).

**Reply by pkoscik** [reply]
Yep, this is my **very** janky approach to this:

https://github.com/pkoscik/EngineeringThesis/blob/main/build.sh
https://github.com/pkoscik/EngineeringThesis/blob/main/poll.sh

`watchexec` detects file changes, then runs the `texlive` in a docker container and sends an update signal to MuPDF

**Comment by astro-ray** [comment]
Is this something that will be implemented in helix? Is there any plan for it?

**Reply by pascalkuthe** [reply]
it already works if you enable the texlab option (which I use regularly, never had any problems). Helix does not and will never contain any language-specific code so there is nothing beyond that will be accepted into core
